src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -0,0 +1,435 @@
+                var examined = readableBuffer.Start;
+                if (!readableBuffer.IsEmpty)
+                {
+                    decoder.Read(readableBuffer);
+                }
+                Input.AdvanceTo(readableBuffer.End);
+            }
+        }
+
+        private async Task SendStreamIdAsync(int id)
+        {
+            await _frameWriter.WriteStreamIdAsync(id);
+        }
+
+        private async Task SendSettingsFrameAsync()
+        {
+            await _frameWriter.WriteSettingsAsync(null);
+        }
+
+        private async Task<Http3ControlStream> CreateNewControlStreamAsync()
+        {
+            var connectionContext = await _http3Connection.Context.ConnectionFeatures.Get<IQuicCreateStreamFeature>().StartUnidirectionalStreamAsync();
+            var httpConnectionContext = new HttpConnectionContext
+            {
+                ConnectionId = connectionContext.ConnectionId,
+                ConnectionContext = connectionContext,
+                Protocols = _context.Protocols,
+                ServiceContext = _context.ServiceContext,
+                ConnectionFeatures = connectionContext.Features,
+                MemoryPool = _context.MemoryPool,
+                Transport = connectionContext.Transport,
+                TimeoutControl = _context.TimeoutControl,
+                LocalEndPoint = connectionContext.LocalEndPoint as IPEndPoint,
+                RemoteEndPoint = connectionContext.RemoteEndPoint as IPEndPoint
+            };
+
+            // TODO think about whether we need a new stream or not here.
+            return new Http3ControlStream(_http3Connection, httpConnectionContext);
+        }
+
+        private async Task<long> TryReadStreamIdAsync()
+        {
+            while (_isClosed == 0)
+            {
+
+                var result = await Input.ReadAsync();
+                var readableBuffer = result.Buffer;
+                var consumed = readableBuffer.Start;
+                var examined = readableBuffer.End;
+
+                try
+                {
+                    if (!readableBuffer.IsEmpty)
+                    {
+                        var id = VariableIntHelper.GetVariableIntFromReadOnlySequence(readableBuffer, out consumed, out examined);
+                        if (id != -1)
+                        {
+                            return id;
+                        }
+                    }
+
+                    if (result.IsCompleted)
+                    {
+                        return -1;
+                    }
+                }
+                finally
+                {
+                    Input.AdvanceTo(consumed, examined);
+                }
+            }
+
+            return -1;
+        }
+
+        public async Task ProcessRequestAsync<TContext>(IHttpApplication<TContext> application)
+        {
+            var streamType = await TryReadStreamIdAsync();
+
+            if (streamType == -1)
+            {
+                // Error dawg
```

Comment:
Make sure you scrub all of these comments.

Reply:
What’s up dawg